README: Document x86-64 requirement

smcv opened the pull request:
Recent versions of the steamwebhelper process
(~/.steam/root/ubuntu12_64/steamwebhelper) are x86-64 executables, and
steamwebhelper is required for a lot of the Steam Client's UI. Many newer
games are also x86-64 executables.

The Steam Client itself and older games continue to be i386 (IA32)
executables, so support for both ABIs is currently necessary.

The 64-bit requirement is difficult to document in terms of Intel brand
names, because most Pentium 4 revisions are i386/IA32-only (more
specifically i686 with SSE2 and sometimes SSE3), but according to
Wikipedia, the Pentium 4 F-series added x86-64/AMD64 support. Pentium D
appears to have been the first Intel brand name that exclusively refers
to x86-64/AMD64 CPUs.

On the AMD side, all Opterons are x86-64/AMD64, so the requirement is
simple to document.

smcv commented:
The 32-bit Pentium 4 models are theoretically meant to be supported, but losing steamwebhelper is fairly fatal to the late 2019 "new UI" (#6795 has a good description of the symptoms of it not working), so this PR updates the README to reflect reality.

> Pentium D appears to have been the first Intel brand name that exclusively refers to x86-64/AMD64 CPUs.

Another possibility would be to document the "official" requirement as "1 GHz Intel Pentium D, AMD Opteron or later with x86-64 (AMD64) instruction set", with the 64-bit Pentium 4 models officially unsupported but unofficially continuing to work fine.

smcv commented on README.md:
```diff
@@ -7,9 +7,12 @@ Hardware and Software Requirements
 Steam for Linux requires the following:
 
-- 1 GHz Pentium 4 or AMD Opteron with 512 megabytes of RAM and 5 gigabytes of hard drive space, or better
+- 1 GHz Pentium 4 or AMD Opteron with x86-64 (AMD64) instruction set
```
According to #4196, x86-64 CPUs without `cmpxchg16b` support (`cx16` in `/proc/cpuinfo`) are also unsupported, but I'm not sure how to describe that requirement in user-facing terms. It's missing from very early AMD x86-64 CPUs, but seems to be implemented in all Intel x86-64 CPUs? The affected CPUs are the same ones that can't run 64-bit Windows 8.1.

kisak-valve commented:
Hello @smcv, #5164 should also be considered if the minimum requirements are bumped.

smcv commented:
> #5164 should also be considered if the minimum requirements are bumped

Yes. SSE2 is part of the x86-64 baseline, but SSE3 isn't.

Is the SSE3 requirement intentional, or is that considered to be a bug?

For what it's worth, x86-64 Linux distributions mostly build their binaries to run on all x86-64 CPUs, and treat unconditional use of SSE3 as a bug (although conditional use with a runtime feature-test is allowed).

smcv commented:
I've added commits to document that CMPXCHG16B and SSE3 are required. The one for SSE3 could be removed or reverted, if the bundled SDL library's SSE3 requirement is considered to be a bug and fixed (it's entirely possible that this has already happened, I don't have a non-SSE3 CPU to test on).

smcv commented:
@kisak-valve, @TTimo: is there anything else blocking this documentation update?
